packages/api/api.go

```diff
@@ -307,6 +317,21 @@ func NewServer(ctx context.Context, drv *sql.Driver, index bool, network string)
 			_, _ = w.Write([]byte(`{"success":true}`))
 		})
 
+		r.HandleFunc("/game/ws", func(w http.ResponseWriter, r *http.Request) {
+			wsConn, err := wsUpgrader.Upgrade(w, r, nil)
+
+			if err != nil {
+				fmt.Errorf("error upgrading websocket: %w", err)
+				return
+			}
+
+			// close the connection when the function returns
+			defer wsConn.Close()
+
+			gameState := game.NewGame(ctx, wsConn)
```

Comment:
I think you want to initialize the game outside of this handler (i.e. before like 320) so the state can be shared across handlers